test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.foreign.allowances
+
+import audit.AuditService
+import base.SpecBase
+import models.UserAnswers
+import models.backend.PropertyDetails
+import org.mockito.ArgumentMatchers.{any, anyString}
+import org.mockito.Mockito.doNothing
+import org.mockito.MockitoSugar.when
+import org.scalatestplus.mockito.MockitoSugar.mock
+import play.api.inject.bind
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import service.{BusinessService, PropertySubmissionService}
+import viewmodels.govuk.SummaryListFluency
+import views.html.foreign.allowances.ForeignAllowancesCheckYourAnswersView
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class ForeignAllowancesCheckYourAnswersControllerSpec extends SpecBase with SummaryListFluency {
+
+  val taxYear = LocalDate.now.getYear
+  val countryCode = "aus"
+
+  val onwardRoute: Call = Call(
+    "GET",
+    s"/update-and-submit-income-tax-return/property/$taxYear/foreign-property/allowances/$countryCode/complete-yes-no"
+  )
+
+  val zeroEmissionCarAllowanceValue = 300
```

Comment:
This should be :

`val zeroEmissionCarAllowance = BigDecimal(300)`

Reply:
its already been converted to BigDecimal in the next line

Reply:
Why we need 2 lines ?